src/main/java/FeatureExtraction/ExtractionAlgorithm.java

```diff
@@ -0,0 +1,7 @@
+package FeatureExtraction;
+
+import java.awt.image.BufferedImage;
+
+public interface ExtractionAlgorithm {
+    Double[] extractFeatures(BufferedImage bufferedImage);
```

Comment:
A dlaczego nie? To się i tak jedno do drugiego konwertuje, a wygodniej czasami używać klasy Double niż typu prymitywnego.